syntax = "proto3";
option java_package = "se.proto";

message SingleIndexPattern {
    string index = 1;
    double lowerBound = 2;
    double upperBound = 3;
}

message SubscribeRequest {
    repeated SingleIndexPattern indexes = 1;
}

message Response {
    enum State {
        BULL = 0;
        BEAR = 1;
    }
    string index = 1;
    double value = 2;
    State state = 3;
}
message AvailableIndexes {
    repeated string indexes = 1;
}
message IndexesRequest {
}


service IndicesService {
    rpc GetStockMarketIndexes (SubscribeRequest) returns (stream Response) {
    }
    rpc GetAvailableIndexes (IndexesRequest) returns (AvailableIndexes) {
    }
}